server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -3470,4 +3464,70 @@ public LiveVersionMap getLiveVersionMap() {
     protected long getPreCommitSegmentGeneration() {
         return preCommitSegmentGeneration.get();
     }
+
+    DirectoryReaderSupplier acquireDirectoryReaderSupplier(SearcherScope scope) throws EngineException {
+        assert scope == SearcherScope.INTERNAL : "acquireDirectoryReaderSupplier(...) isn't prepared for external usage";
+
+        /* Acquire order here is store -> manager since we need
+         * to make sure that the store is not closed before
+         * the searcher is acquired. */
+        if (store.tryIncRef() == false) {
```

Comment:
Ah see @tlrx, we don't need to ever acquire a store because we are guaranteed to have an engine reference here?

Reply:
I see, so we can just change that into an assertion?

Reply:
I think so yes, assertion should be enough :)

Reply:
I think so too

Reply:
pushed: b81fa738b37df2452430ee34c6288d811ccf64da